src/java.base/share/classes/jdk/internal/util/ArraysSupport.java

```diff
@@ -575,52 +575,84 @@ public static int mismatch(double[] a, int aFromIndex,
      */
-    public static final int MAX_ARRAY_LENGTH = Integer.MAX_VALUE - 8;
+    public static final int SOFT_MAX_ARRAY_LENGTH = Integer.MAX_VALUE - 8;
 
     /**
-     * Calculates a new array length given an array's current length, a preferred
-     * growth value, and a minimum growth value.  If the preferred growth value
-     * is less than the minimum growth value, the minimum growth value is used in
-     * its place.  If the sum of the current length and the preferred growth
-     * value does not exceed {@link #MAX_ARRAY_LENGTH}, that sum is returned.
-     * If the sum of the current length and the minimum growth value does not
-     * exceed {@code MAX_ARRAY_LENGTH}, then {@code MAX_ARRAY_LENGTH} is returned.
-     * If the sum does not overflow an int, then {@code Integer.MAX_VALUE} is
-     * returned.  Otherwise, {@code OutOfMemoryError} is thrown.
+     * Computes a new array length given an array's current length, a minimum growth
+     * amount, and a preferred growth amount. The computation is done in an overflow-safe
+     * fashion.
+     *
+     * This method is used by objects that contain an array that might need to be grown
+     * in order to fulfill some immediate need (the minimum growth amount) but would also
+     * like to request more space (the preferred growth amount) in order to accommodate
+     * potential future needs. The returned length is usually clamped at the soft maximum
+     * length in order to avoid hitting the JVM implementation limit. However, the soft
+     * maximum will be exceeded if the minimum growth amount requires it.
+     *
+     * If the preferred growth amount is less than the minimum growth amount, the
+     * minimum growth amount is used as the preferred growth amount.
+     *
+     * The preferred length is determined by adding the preferred growth amount to the
+     * current length. If the preferred length does not exceed the soft maximum length
+     * (SOFT_MAX_ARRAY_LENGTH) then the preferred length is returned.
      *
-     * @param oldLength   current length of the array (must be non negative)
-     * @param minGrowth   minimum required growth of the array length (must be
-     *                    positive)
-     * @param prefGrowth  preferred growth of the array length (ignored, if less
-     *                    then {@code minGrowth})
+     * Since the preferred length exceeds the soft maximum, we use the minimum growth
+     * amount. The minimum required length is determined by adding the minimum growth
+     * amount to the current length. If the minimum required length is less than the
+     * soft maximum, the soft maximum is returned. If the minimum required length is
+     * greater than the soft maximum but does not exceed Integer.MAX_VALUE, the minimum
+     * required length is returned. Otherwise, the minimum required length exceeds
+     * Integer.MAX_VALUE, which can never be fulfilled, so this method throws OutOfMemoryError.
```

Comment:
I think you can simplify with:

```suggestion
     * If the preferred length exceeds the soft maximum, we use the minimum growth
     * amount. The minimum required length is determined by adding the minimum growth
     * amount to the current length. 
     * If the minimum required length exceeds Integer.MAX_VALUE, then this method 
     * throws OutOfMemoryError. Otherwise, this method returns the soft maximum or
     * minimum required length, which ever is greater.
```

Then i think it follows that `Math.max` can be used in the implementation.

Reply:
I agree with Paul's comment of Dec 8.  Especially the s/Since/If/

I would amend with s/which ever/whichever/ but that might be my dialect of North American